pilot/pkg/proxy/envoy/v1/discovery.go

```diff
@@ -469,6 +469,12 @@ func (ds *DiscoveryService) ClearCacheStats(_ *restful.Request, _ *restful.Respo
 	ds.ldsCache.resetStats()
 }
 
+// ClearCache is wrapper for clearCache method, used when new controller gets
+// instantiated dynamically
+func (ds *DiscoveryService) ClearCache() {
+	ds.clearCache()
+}
+
```

Comment:
consider moving line 353-360 out to a new external method, and have the secret handler calls that method.

Reply:
@baodongli if there is no technical objection, I would prefer at least for now to leave as is, since it is easier to understand what the code does, but in the followup PR it can be refactored.